### PR TITLE
Particlenet added and all km3net dependencies are deleted

### PR DESCRIPTION
**Pull request by giogiopg**
ParticleNeT model for Jet Tagging with particle cloud-like data used in KM3NeT was successfully implemented in GraphNeT. It uses all the utilities from GraphNeT, mainly DynEdgeConv blocks, to build its architecture. References:

- https://arxiv.org/pdf/1902.08570
- https://ecap.nat.fau.de/wp-content/uploads/2023/06/2023-06_LukasHennig_MSc_ORCA_TauID_GNNs.pdf

Pull request previously opened by @ivanMM27 with the difference that all the km3net data processing dependencies have been removed.

### CODE REVIEW

File: src/graphnet/models/gnn/ParticleNeT.py [hunk 1]
```diff
@@ -0,0 +1,255 @@
...
+            readout_layer_sizes: Hidden layer size in the MLP following the
+                post-processing _and_ optional global pooling. As this is the
+                last layer in the model, it yields the output of the `DynEdge`
+                model. Defaults to [256,].
+            global_pooling_schemes: The list global pooling schemes to use.
+                Options are: "min", "max", "mean", and "sum".
+                Default to "mean".
+            activation_layer: The activation function to use in the model.
+                Default to "relu".
+            add_batchnorm_layer: Whether to add a batch normalization layer after
```
Comment (RasmusOrsoe) [inline src/graphnet/models/gnn/ParticleNeT.py:75]: Line too long. See CodeClimate.

File: src/graphnet/models/gnn/ParticleNeT.py [hunk 2]
```diff
@@ -0,0 +1,255 @@
...
+                each linear layer. Default to True.
+            dropout_readout: Dropout value to use in the readout layer(s).
+                Default to 0.1.
+            skip_readout: Whether to skip the readout layer(s). If `True`, the
+                output of the last DynEdgeConv block is returned directly.
+        """
+        # Latent feature subset for computing nearest neighbours in ParticleNeT.
```
Comment (RasmusOrsoe) [inline src/graphnet/models/gnn/ParticleNeT.py:82]: Line too long. See CodeClimate

File: src/graphnet/models/gnn/__init__.py [hunk 3]
```diff
@@ -6,3 +6,4 @@
 from .dynedge_kaggle_tito import DynEdgeTITO
 from .RNN_tito import RNN_TITO
 from .icemix import DeepIce
+from .ParticleNeT import ParticleNeT
```
Comment (RasmusOrsoe) [inline src/graphnet/models/gnn/__init__.py:9]: `from .particlenet import ParticleNeT`

**Review by RasmusOrsoe — COMMENTED** [review]
Hey @giogiopg @IvanMM27 - thanks for this very clean pull request 🚀 !

There is only a few minor changes needed before we can merge this. 

1. `ParticleNeT.py` should be changed to `particlenet.py` or `particle_net.py` so that it is in line with the snake case convention. (remember to adjust the import statement in the `__init__.py` file accordingly)
2. Two lines exceed the maximum line length (see comments)

**Comment by IvanMM27** [comment]
Hello @RasmusOrsoe, I made the changes you commented. Nevertheless, the codeclimate scan shows 6 errors to fix that I think they were previously too. Could you please review this? Thank you!

Adding @giogiopg for completeness

**Comment by RasmusOrsoe** [comment]
@IvanMM27 & @giogiopg  - looks like the new changes make some of the checks fail.  Could you take a look at that?

**Review by RasmusOrsoe — APPROVED** [review]
:rocket:

**Pull request merged**